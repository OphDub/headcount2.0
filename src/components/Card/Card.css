article {
    border: 1px solid lightslategray;
    height: 325px;
    width: 200px;
    margin: 10px;
    font-family: 'Francois One', sans-serif;
}

ul {
    margin: auto;
    padding-left: 60px;
    list-style-type: none;
}

h4 {
    margin: auto;
    margin-bottom: 20px;
    padding: 10px;
    text-align: center;
    background-color: rgb(74, 169, 241);
    color: rgb(232, 238, 241);
}

.highFive {
    color: blue;
}

.lowFive {
    color: orangered;
}

.cardContainer article:hover {
    box-shadow: 4px 4px 0px 0px rgba(22, 22, 87, 0.2);
}

.compareContainer.selected {
  border: 3px solid rgb(57, 185, 231);
}

.selected {
    border: 3px solid rgb(57, 185, 231);
}