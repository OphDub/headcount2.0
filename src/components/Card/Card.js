import React from 'react';
import { string, object } from 'prop-types';
import './Card.css';

const Card = ({ location, data }) => {
  // <article onClick={() => selectCard(location)}> ?
  const eachCard = Object.keys(data).map( dataPoint => {
                                            const highOrLow = data[ dataPoint ] >= 0.5 ? "highFive" : "lowFive";
                                            
                                            return <li className={ highOrLow } >{ dataPoint + ': ' + data[ dataPoint ] }</li>
                                        } )
  return (
    <article>
      <h4>{ location }</h4>
      <ul>{ eachCard }</ul>
    </article>
  )
}

Card.propTypes = {
  location: string.isRequired,
  data: object.isRequired,
}

export default Card;
