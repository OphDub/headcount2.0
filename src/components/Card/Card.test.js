import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('CARD', () => {
  let wrapper
  let mockId
  let mockLocation
  let mockData
  let mockFn

  beforeEach(() => {
    mockId = 'HOLLERADO'
    mockLocation = 'COLORADO'
    mockData = { 2004: 0.4, 2005: 0.5, 2006: 0.75, 2007: 0.8 }
    mockFn = jest.fn()
    wrapper = shallow (<Card  id={mockId}
                              location={mockLocation}
                              data={mockData}
                              selectCard={mockFn}/>)
  })

  it('it should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a location an an array of data points', () => {
    expect(wrapper.find('h4').text()).toEqual(mockLocation)
    expect(wrapper.find('ul').children().length).toEqual(4)
  })

  it('should have specific styling on list items depending on their value', () => {
    expect(wrapper.find('ul').children().first().hasClass('lowFive')).toEqual(true)
    expect(wrapper.find('ul').children().last().hasClass('highFive')).toEqual(true)
  })

  it('should call the selectCard method on click', () => {
    wrapper.simulate('click')

    expect(mockFn.mock.calls.length).toEqual(1)
    expect(mockFn.mock.calls).toEqual([[mockId]]);
  })
})