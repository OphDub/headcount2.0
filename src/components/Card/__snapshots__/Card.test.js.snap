// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD it should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "2004": 0.4,
            "2005": 0.5,
            "2006": 0.75,
            "2007": 0.8,
          }
    }
    id="HOLLERADO"
    location="COLORADO"
    selectCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          COLORADO
</h4>,
        <ul>
          <li
                    className="lowFive"
          >
                    2004: 0.4
          </li>
          <li
                    className="highFive"
          >
                    2005: 0.5
          </li>
          <li
                    className="highFive"
          >
                    2006: 0.75
          </li>
          <li
                    className="highFive"
          >
                    2007: 0.8
          </li>
</ul>,
      ],
      "className": undefined,
      "id": "COLORADO",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="lowFive"
>
              2004: 0.4
</li>,
            <li
              className="highFive"
>
              2005: 0.5
</li>,
            <li
              className="highFive"
>
              2006: 0.75
</li>,
            <li
              className="highFive"
>
              2007: 0.8
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2004: 0.4",
              "className": "lowFive",
            },
            "ref": null,
            "rendered": "2004: 0.4",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2005: 0.5",
              "className": "highFive",
            },
            "ref": null,
            "rendered": "2005: 0.5",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2006: 0.75",
              "className": "highFive",
            },
            "ref": null,
            "rendered": "2006: 0.75",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2007: 0.8",
              "className": "highFive",
            },
            "ref": null,
            "rendered": "2007: 0.8",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            COLORADO
</h4>,
          <ul>
            <li
                        className="lowFive"
            >
                        2004: 0.4
            </li>
            <li
                        className="highFive"
            >
                        2005: 0.5
            </li>
            <li
                        className="highFive"
            >
                        2006: 0.75
            </li>
            <li
                        className="highFive"
            >
                        2007: 0.8
            </li>
</ul>,
        ],
        "className": undefined,
        "id": "COLORADO",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="lowFive"
>
                2004: 0.4
</li>,
              <li
                className="highFive"
>
                2005: 0.5
</li>,
              <li
                className="highFive"
>
                2006: 0.75
</li>,
              <li
                className="highFive"
>
                2007: 0.8
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2004: 0.4",
                "className": "lowFive",
              },
              "ref": null,
              "rendered": "2004: 0.4",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2005: 0.5",
                "className": "highFive",
              },
              "ref": null,
              "rendered": "2005: 0.5",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2006: 0.75",
                "className": "highFive",
              },
              "ref": null,
              "rendered": "2006: 0.75",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2007: 0.8",
                "className": "highFive",
              },
              "ref": null,
              "rendered": "2007: 0.8",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
